### PR TITLE
Content stream feature

### PR DESCRIPTION
**Pull request by patrickmann**
APIs to support the new Content Stream feature Graylog2/graylog-plugin-enterprise#5404

/jpd Graylog2/graylog-plugin-enterprise#5647

### CODE REVIEW

File: graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config [hunk 1]
```diff
@@ -66,4 +66,4 @@ cloud_inputs=on
 preflight_web=off
 
 # Instant archiving is disabled by default
-instant_archiving=off
+instant_archiving=on
```
Comment (moesterheld) [inline graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config:69]: did this one slip into this PR?

File: graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResource.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import java.util.List;
+
+import static org.graylog2.shared.rest.documentation.generator.Generator.CLOUD_VISIBLE;
+
+@RequiresAuthentication
+@Api(value = "ContentStream", description = "Content Stream", tags = {CLOUD_VISIBLE})
+@Path("/contentStream/")
+@Produces(MediaType.APPLICATION_JSON)
+//@Consumes(MediaType.APPLICATION_JSON)
+public class ContentStreamResource extends RestResource {
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResource.java:49]: Is there deliberately no permission scheme in place here? Any user can change/read the settings of other users. 

I think that is not intended.

File: graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResource.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            @ApiParam(name = "username") @PathParam("username") String username
+    ) throws NotFoundException {
+        return getSettings(username);
+    }
+
+    @PUT
+    @Path("enable/{username}")
+    @ApiOperation("Enable Content Stream for specified user")
+    @AuditEvent(type = AuditEventTypes.CONTENT_STREAM_USER_SETTINGS_UPDATE)
+    public void contentStreamEnable(
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResource.java:77]: The update methods could return the updated `ContentStreamUserSettings` object. I think most other calls in Graylog do.

File: graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResponseFactory.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.contentStream.rest;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+class ContentStreamResponseFactory {
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/contentStream/rest/ContentStreamResponseFactory.java:22]: Class isn't used

File: graylog2-server/src/main/java/org/graylog2/contentStream/db/ContentStreamUserSettingsDto.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import com.google.auto.value.AutoValue;
+import org.mongojack.Id;
+import org.mongojack.ObjectId;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoValue
+public abstract class ContentStreamUserSettingsDto {
```
Comment (moesterheld) [inline graylog2-server/src/main/java/org/graylog2/contentStream/db/ContentStreamUserSettingsDto.java:30]: the `Dto` is a not really correct here as this is the persisted entity

**Review by moesterheld — CHANGES_REQUESTED** [review]
Works as expected, but I have a few naming convention nitpicks.

Other than that, I am worried that there are no permission checks in place.

File: graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config (hunk 1 above)
Comment (patrickmann) [reply graylog2-server/src/main/resources/org/graylog2/featureflag/feature-flag.config]: Good catch - no wonder IT-tests were failing.

File: graylog2-web-interface/src/components/content-stream/ContentStreamReleasesSection.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import { RelativeTime, Spinner } from 'components/common';
+import useContentStream from 'components/content-stream/hook/useContentStream';
+
+type Props = {
+  rssUrl: string,
+};
+
+const StyledListGroupItem = styled(ListGroupItem)`
+  display: flex;
+  gap: 16px;
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/ContentStreamReleasesSection.tsx:33]: Can we use theme spacing here?

File: graylog2-web-interface/src/components/content-stream/ContentStreamReleasesSection.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    return <Spinner />;
+  }
+
+  return (
+    <ListGroup>
+      {feedList.map((feed) => (
+        <StyledListGroupItem key={feed?.guid['#text'] || feed?.title}>
+          <a href={feed?.link} target="_blank" rel="noreferrer">
+            {/* eslint-disable-next-line react/no-danger */}
+            <span dangerouslySetInnerHTML={{ __html: _sanitizeText(feed.title) }} />
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/ContentStreamReleasesSection.tsx:60]: Is it intentional to not add the `?` for `feed` like the other instances (`feed?.title`) ?

File: graylog2-web-interface/src/components/content-stream/hook/useContentStream.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import { useQuery } from '@tanstack/react-query';
+import { XMLParser } from 'fast-xml-parser';
+
+import UserNotification from 'preflight/util/UserNotification';
+
+export type FeedITem = {
+  title: string,
+  link: string,
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/hook/useContentStream.ts:24]: Do some of the fields need an `| undefined` ?

File: graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+type Props = {
+  feed: FeedITem
+}
+
+const StyledImage = styled.img`
+  max-width: 100%;
+  width: 100%;
+  object-fit: contain;
+  border-radius: 4px 0;
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx:34]: Maybe we can use theme spacing here too?

File: graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+  flex-grow: 1;
+  background-color: ${theme.colors.table.backgroundAlt};
+
+  > a {
+    font-weight: bold;
+  }
+`);
+const StyledPanelFooter = styled(Panel.Footer)(({ theme }: { theme: DefaultTheme }) => css`
+  background-color: ${theme.colors.table.backgroundAlt};
+  border-radius: 0 0 4px 4px;
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx:46]: Same here

File: graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+`);
+const StyledPanel = styled(Panel)`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  border: none;
+  border-radius: 4px;
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx:53]: and here

File: graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+`;
+const _sanitizeText = (text) => DOMPurify.sanitize(text);
+
+const ContentStreamNewsItem = ({ feed }: Props) => (
+  <Carousel.Slide>
+    <StyledPanel>
+      <StyledImage src={feed?.['media:content']?.attr_url} alt={feed?.title} />
+      <StyledPanelBody>
+        {/* eslint-disable-next-line react/no-danger */}
+        <a href={feed.link} target="_blank" rel="noreferrer"><span dangerouslySetInnerHTML={{ __html: _sanitizeText(feed?.title) }} />
```
Comment (gally47) [inline graylog2-web-interface/src/components/content-stream/news/ContentStreamNewsItem.tsx:63]: Is it intentional to not add the `?` for `feed` like the other instances (`feed?.link`) ?

**Review by gally47 — COMMENTED** [review]
Code Looks good! Found some small nitpicks/questions below.
Will be testing the UI next.

**Review by gally47 — APPROVED** [review]
Tested UI, LGTM!

**Comment by gally47** [comment]
Tested new changes. LGTM!

**Review by gally47 — APPROVED** [review]

**Pull request merged**